packages/stream_chat_flutter_core/lib/src/stream_chat_core.dart

```diff
@@ -113,11 +109,80 @@ class StreamChatCoreState extends State<StreamChatCore>
   var _isInForeground = true;
   var _isConnectionAvailable = true;
 
+  Future<SystemEnvironment> get _getSystemEnvironment async {
+    String? osVersion;
+    String? deviceModel;
+
+    try {
+      switch (CurrentPlatform.type) {
```

Comment:
Following my other comment, I would move this to an `AppConfig` class which can be used both by core and flutter.

Reply:
We can move the fields in the AppConfig class but we can't move this logic as it's flutter specific and we can't add flutter related dependency in llc